### PR TITLE
use repr(fty) for enums

### PR DESCRIPTION
**Pull request by burrbull**
enums are now in C-style with `#repr(..)`

Should slightly reduce code size in debug mode when enums are used.

r? @Disasm
cc @therealprof

### CODE REVIEW

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
As `msp430g2553` contains empty `<enumeratedValues/>`, we should use new `svd-parser` with https://github.com/rust-embedded/svd/pull/69 patch.

**Comment by burrbull** [comment]
bors try

**Comment by Emilgardis** [comment]
Why retry? Is error E0084 not valid?

**Comment by burrbull** [comment]
Because it should be fixed in #372 .

@Emilgardis What do you think about topic? Is it right way?

**Comment by burrbull** [comment]
Change example:
изображение

**Comment by therealprof** [comment]
Uhm, which MCUs are affected by the change? I just tried it on a STM32F042 but there's literally no change.

**Comment by burrbull** [comment]
Rebased. `repr` attribute removed for `bool` fields.
bors try

**Comment by Emilgardis** [comment]
I'm not sure I like this solution. I don't have anything against merging it, but it just seems weird.

**Comment by burrbull** [comment]
> but it just seems weird.

What exactly do you not like? `repr` attributes?

**Comment by Emilgardis** [comment]
My bad, I thought we already did this, misunderstood what this pr does. This is very sensible to do

File: src/generate/register.rs [hunk 1]
```diff
@@ -636,14 +636,23 @@ impl Variant {
 fn add_from_variants(mod_items: &mut Vec<TokenStream>, variants: &Vec<Variant>, pc: &Ident, f: &F, desc: &str, reset_value: Option<u64>) {
     let fty = &f.ty;
 
+    let mut repr = quote! { #[repr(#fty)] };
+    let mut cast = quote! { variant as _ };
+
+    if f.ty == "bool" {
+        repr = quote! { };
+        cast = quote! { variant as u8 != 0 };
+    }
```
Comment (Emilgardis) [inline src/generate/register.rs]: Nitpick: This could be if else instead, have no real preference however.

Comment (burrbull) [reply src/generate/register.rs]: Done

**Pull request merged**